src/sas/qtgui/Utilities/Preferences/FittingOptionsWidget.py

```diff
@@ -0,0 +1,547 @@
+import os
+import types
+import webbrowser
+
+from bumps.options import FIT_CONFIG, FIT_FIELDS, ChoiceList
+
+from PyQt5 import QtCore
+from PyQt5 import QtGui
+from PyQt5 import QtWidgets
+
+from sas.system import config
+
+import sas.qtgui.Utilities.GuiUtils as GuiUtils
+from sas.qtgui.Utilities.Preferences.PreferencesWidget import PreferencesWidget, set_config_value
+
+
+class FittingOptions(PreferencesWidget):
+    """
+    Hard-coded version of the fit options dialog available from BUMPS.
+    bumps.options.FIT_FIELDS gives mapping between parameter names, parameter strings and field type
+    (double line edit, integer line edit, combo box etc.), e.g.::
+
+        FIT_FIELDS = dict(
+            samples = ("Samples", parse_int),
+            xtol = ("x tolerance", float),
+        )
+
+    bumps.fitters.<algorithm>.settings gives mapping between algorithm, parameter name and default value, e.g.::
+
+        >>> settings = [('steps', 1000), ('starts', 1), ('radius', 0.15), ('xtol', 1e-6), ('ftol', 1e-8)]
+    """
+    fit_option_changed = QtCore.pyqtSignal(str)
+
+    name = "Fitting Optimizers"
+
+    def __init__(self):
+        super(FittingOptions, self).__init__(self.name)
+
+    def _addAllWidgets(self):
+        # Add default fit algorithm widget
+        self.config = FIT_CONFIG
+        active_names = [name for fit_id, name in self.config.names.items() if fit_id in self.config.active_ids]
+        default_id = config.DEFAULT_FITTING_OPTIMIZER
+        algorithm = self.config.names[default_id]
+        self.addHeaderText("Values to persist between SasView sessions:")
+        self.defaultOptimizer = self.addComboBox("Default Fit Algorithm",
+                                                 active_names,
+                                                 self.setDefaultOptimizer,
+                                                 algorithm)
+        self.addHorizontalLine()
+
+        # Add all other widgets
+        self.addHeaderText("Fitting options for this session only:")
+        self.cbAlgorithm = self.addComboBox("Active Fit Algorithm",
+                                            active_names,
+                                            self.setActiveOptimizer,
+                                            algorithm)
+        self._addFittingOptionsWidgets()
+        self.verticalLayout.addWidget(self.stackedWidget)
+        self.onAlgorithmChange(self.cbAlgorithm.currentIndex())
+
+    def _addFittingOptionsWidgets(self):
+        """This is a direct copy from teh python file generated from the old UI file."""
```

Comment:
Would it be very annoying to have the UI element definitions and layouts in a separate file, preferably a .ui file from the Designer? Or if not, maybe at least a separate .py file/class with those elements, so there's a clear distinction between UI and logic code?

Reply:
@pkienzle had a similar complaint with this change but more related to pulling the widget out of the fitting perspective, breaking the perspective as a plugin framework. I might merge this as-is, document all issues in #2316, and handle these issues one by one in separate branches.